packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/model/clientmodel/ClientMethod.java

```diff
@@ -511,6 +511,16 @@ public void addImportsTo(Set<String> imports, boolean includeImplementationImpor
                 ClassType.BINARY_DATA.addImportsTo(imports, false);
                 ClassType.HTTP_REQUEST.addImportsTo(imports, false);
             }
+
+            // sync-stack, lro (+ pageable)
+            boolean isLroPageable = (type == ClientMethodType.PagingSyncSinglePage
+                && proxyMethod != null
+                && GenericType.Response(ClassType.BINARY_DATA).equals(proxyMethod.getReturnType().getClientType()));
```

Comment:
trying to learn the code - is it the case that when a code-model operation has LRO extension applied, the sync-proxy-method for that operation will return `Response<BinaryData>`  ?

Reply:
Yes. 
Reason we use BinaryData instead of concrete model itself, is due to serialization/deserialization of read-only properties. See [discussion](https://github.com/Azure/autorest.java/issues/2284#issuecomment-2588682428).

For async-proxy-method, it's `Mono<Response<Flux<ByteBuffer>>>`.